subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -488,7 +488,6 @@ static int usbd_cdc_acm_init(struct usbd_class_data *const c_data)
 	struct cdc_acm_uart_data *data = dev->data;
 	struct usbd_cdc_acm_desc *desc = data->desc;
 
-	desc->iad.bFirstInterface = desc->if0.bInterfaceNumber;
```

Comment:
>This fixes UAC2 instances where the IAD bFirstInterface is not 0, e.g.
when HID was used together with UAC2.

Okay to rework it, but I did not understand what it actually fixes. When usbd_class_init() is called, desc->if0.bInterfaceNumber should have an updated interface number.

Reply:
UAC2 doesn't have (never had) the corresponding line to set bFirstInterface based on bInterfaceNumber.

This PR makes all classes to have the bFirstInterface set with common code.